src/analysisd/rules.c

```diff
@@ -118,20 +118,43 @@ int Rules_OP_ReadRules(const char *rulefile)
     const char *xml_same_source_ip = "same_source_ip";
     const char *xml_same_src_port = "same_src_port";
     const char *xml_same_dst_port = "same_dst_port";
+    const char *xml_same_srcuser = "same_src_user";
     const char *xml_same_user = "same_user";
     const char *xml_same_location = "same_location";
     const char *xml_same_id = "same_id";
     const char *xml_dodiff = "check_diff";
     const char *xml_same_field = "same_field";
+    const char *xml_same_destination_ip = "same_destination_ip";
+    const char *xml_same_agent = "same_agent";
+    const char *xml_same_url = "same_url";
+    const char *xml_same_srcgeoip = "same_srcgeoip";
+    const char *xml_same_protocol = "same_protocol";
+    const char *xml_same_action = "same_action";
+    const char *xml_same_data = "same_data";
+    const char *xml_same_extra_data = "same_extra_data";
+    const char *xml_same_status = "same_status";
+    const char *xml_same_system_name = "same_system_name";
+    const char *xml_same_dstgeoip = "same_dstgeoip";
 
     const char *xml_different_url = "different_url";
-    const char *xml_different_srcip = "different_srcip";
+    const char *xml_different_srcip = "different_source_ip";
     const char *xml_different_srcgeoip = "different_srcgeoip";
+    const char *xml_different_destination_ip = "different_destination_ip";
+    const char *xml_different_src_port = "different_src_port";
+    const char *xml_different_dst_port = "different_dst_port";
+    const char *xml_different_location = "different_location";
+    const char *xml_different_protocol = "different_protocol";
+    const char *xml_different_action = "different_action";
+    const char *xml_different_srcuser = "different_src_user";
+    const char *xml_different_user = "different_user";
+    const char *xml_different_id = "different_id";
+    const char *xml_different_data = "different_data";
+    const char *xml_different_extra_data = "different_extra_data";
+    const char *xml_different_status = "different_status";
+    const char *xml_different_system_name = "different_system_name";
+    const char *xml_different_dstgeoip = "different_dstgeoip";
+    const char *xml_different_agent = "different_agent";
 
-    const char *xml_notsame_source_ip = "not_same_source_ip";
```

Comment:
We must not delete the `not_same` options, because that would make the parser fail for rule definitions that already use them.
Let's restore them, and join them to their equivalent `different_*` option.